qunicorn_core/core/job_service.py

```diff
@@ -135,3 +135,32 @@ def delete_jobs_by_deployment_id(deployment_id, user_id: Optional[str] = None) -
     jobs = get_jobs_by_deployment_id(deployment_id, user_id=user_id)
     job_db_service.delete_jobs_by_deployment_id(deployment_id)
     return jobs
+
+
+def get_job_queue_items() -> dict:
+    """Get the latest running job and all latest ready jobs"""
+    all_jobs: list[JobDataclass] = job_db_service.get_all()
+    return {"running_job": get_latest_running_job(all_jobs), "queued_jobs": get_latest_ready_jobs(all_jobs)}
+
+
+def get_latest_ready_jobs(all_jobs: list[JobDataclass]) -> list[SimpleJobDto]:
+    """Get all latest jobs with state ready until reaching the first finished job"""
+    all_jobs.sort(reverse=True, key=lambda j: j.id)
+    latest_jobs: list[SimpleJobDto] = []
+    for job in all_jobs:
+        if job.state == JobState.READY:
+            latest_jobs.append(job_mapper.dataclass_to_simple(job))
+        elif job.state == JobState.FINISHED:
+            break
+    latest_jobs.sort(key=lambda j: j.id)
+    return latest_jobs
+
+
+def get_latest_running_job(all_jobs: list[JobDataclass]) -> SimpleJobDto | None:
```

Comment:
Same as for ready jobs. I assume a method to show all running jobs would be more useful.